addons/sale_purchase/models/product_template.py

```diff
@@ -9,23 +9,26 @@ class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
     service_to_purchase = fields.Boolean(
-        "Subcontract Service",
-        compute='_compute_service_to_purchase', store=True, readonly=False,
+        "Subcontract Service", company_dependent=True,
         help="If ticked, each time you sell this product through a SO, a RfQ is automatically created to buy the product. Tip: don't forget to set a vendor on the product.")
 
-    _sql_constraints = [
-        ('service_to_purchase', "CHECK((type != 'service' AND service_to_purchase != true) or (type = 'service'))", 'Product that is not a service can not create RFQ.'),
-    ]
-
-    @api.constrains('service_to_purchase', 'seller_ids')
+    @api.constrains('service_to_purchase', 'seller_ids', 'type')
     def _check_service_to_purchase(self):
         for template in self:
-            if template.service_to_purchase and not template.seller_ids:
-                raise ValidationError(_(
-                    "Please define the vendor from whom you would like to purchase this service automatically."))
+            if template.service_to_purchase:
+                if template.type != 'service':
+                    raise ValidationError(_("Product that is not a service can not create RFQ."))
+                elif not template.seller_ids:
+                    raise ValidationError(_("Please define the vendor from whom you would like to purchase this service automatically."))
 
-    @api.depends('type', 'expense_policy')
-    def _compute_service_to_purchase(self):
-        for template in self:
-            if template.type != 'service' or template.expense_policy != 'no':
-                template.service_to_purchase = False
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('service_to_purchase') and not vals.get('seller_ids'):
+                raise ValidationError(_("Please define the vendor from whom you would like to purchase this service automatically."))
+        return super().create(vals_list)
+
+    @api.onchange('type', 'expense_policy')
+    def _onchange_service_to_purchase(self):
+        products_template = self.filtered(lambda p: p.type != 'service' or p.expense_policy != 'no')
+        products_template.service_to_purchase = False
```

Comment:
@IT-Ideas  I don't think company-dependent field support compute on the field that's why I use onchange instead of compute for ref. https://github.com/odoo/odoo/blob/master/odoo/models.py#L4141-L4149. Thanks

Reply:
What about extracting the compute computation into an external def, use it in the compute (by passing the values from the record) as well as in the the create (by passing the value from the vals). That way we would prevent falling in the value = default case when not wanted, no?

Reply:
Hello @hiroh-odoo Did you get what I proposed here above? Keeping the _compute calling and external def that would also be used in the create. Wouldn't that work?

Reply:
Hello @IT-Ideas , Now the `service_to_purchase` is the `company_dependent` field so we can not make that field as compute. if we want to make it compute, then we have to remove the `company_dependent` attribute and need to handle all property values through the compute and inverse manually, which is not preferable.

So we have removed compute and added the onchange. Second, constraint was on company_dependent(non-stored) field so it was not fired at the time of the creation of a record that's why we checked it manually while creating a record. 
What do you suggest?

Thanks

Reply:
Ok I understand but I am still concerned that this would be changed through python code one day. But I don't have better idea.